static/js/lib/redux_api.js

```diff
@@ -0,0 +1,166 @@
+// @flow
+import { createAction } from 'redux-actions';
+import type { Dispatch } from 'redux';
+import R from 'ramda';
+
+import type { Action } from '../flow/reduxTypes';
+import type { Endpoint, RestState } from '../flow/restTypes';
+import { fetchJSONWithCSRF } from './api';
+import {
+  FETCH_PROCESSING,
+  FETCH_SUCCESS,
+  FETCH_FAILURE,
+} from '../actions';
+import { GET, PATCH, POST } from '../constants';
+import { couponEndpoint } from '../reducers/coupons';
+
+const actionize = R.compose(R.toUpper, R.join("_"));
+
+export const requestActionType = (...xs: string[]) => `REQUEST_${actionize(xs)}`;
+
+export const successActionType = (...xs: string[]) => `RECEIVE_${actionize(xs)}_SUCCESS`;
+
+export const failureActionType = (...xs: string[]) => `RECEIVE_${actionize(xs)}_FAILURE`;
+
+export const clearActionType = (...xs: string[]) => `CLEAR_${actionize(xs)}`;
+
+export const INITIAL_STATE: RestState = {
+  loaded: false,
+  processing: false,
+};
+
+const defaultRESTPrefixes = { GET, PATCH, POST };
+
+const getPrefixForEndpoint = (verb, endpoint) => (
+  R.propOr(defaultRESTPrefixes[verb], `${R.toLower(verb)}Prefix`, endpoint)
+);
+
+export function makeFetchFunc(endpoint: Endpoint): (...args: any) => Promise<*> {
+  return (...args) => fetchJSONWithCSRF(endpoint.url, endpoint.makeOptions(...args));
```

Comment:
Oh right I forgot about that, do you think we should just go with what we discussed above?